instance/models/mixins/storage.py

```diff
@@ -109,3 +123,116 @@ def deprovision_swift(self):
                     self.logger.exception('Could not delete Swift container "%s".', container_name)
             self.swift_provisioned = False
             self.save()
+
+
+class S3BucketInstanceMixin(models.Model):
+    """
+    Mixin to provision S3 bucket for an instance.
+    """
+    class Meta:
+        abstract = True
+
+    def get_s3_policy(self):
+        """
+        Return s3 policy with access to create and update bucket
+        """
+        policy = {
+            "Version": "2012-10-17",
+            "Statement": [
+                {
+                    "Effect": "Allow",
+                    "Action": ["s3:ListBucket", "s3:CreateBucket", "s3:DeleteBucket"],
+                    "Resource": ["arn:aws:s3:::{}".format(self.s3_bucket_name)]
+                },
+                {
+                    "Effect": "Allow",
+                    "Action": [
+                        "s3:*Object*"
+                    ],
+                    "Resource": ["arn:aws:s3:::{}/*".format(self.s3_bucket_name)]
+                }
+            ]
+        }
+        return json.dumps(policy, indent=2)
+
+    @property
+    def bucket_name(self):
+        """
+        Return bucket name truncated to 50 characters
+        """
+        return truncate_name(
+            '{}-{}'.format(settings.S3_BUCKET_PREFIX, self.database_name.replace('_', '-')),
+            length=50
```

Comment:
I think buckets can be [63 characters long](https://docs.aws.amazon.com/awscloudtrail/latest/userguide/cloudtrail-s3-bucket-naming-requirements.html)?